baseplate/secrets/fetcher.py

```diff
@@ -282,13 +282,19 @@ def is_about_to_expire(self):
     def get_secret(self, secret_name):
         """Get the value and expiration time of a named secret."""
         logger.debug("Fetching secret %r.", secret_name)
-        response = self.session.get(
-            urljoin(self.base_url, posixpath.join("v1", secret_name)),
-            headers={
-                "X-Vault-Token": self.token,
-            },
-        )
-        response.raise_for_status()
+        try:
+            response = self.session.get(
+                urljoin(self.base_url, posixpath.join("v1", secret_name)),
+                headers={
+                    "X-Vault-Token": self.token,
+                },
+            )
+            response.raise_for_status()
+        except requests.HTTPError:
+            logger.exception("Vault client failed to get secret: %s",
```

Comment:
can you switch this to `logger.error` so you're not duplicate logging the stack trace (it'll happen here and also when the exception is raised)

Reply:
it might also be nice to include the status code or error message from the response if one exists.